lib/diff.mli

```diff
@@ -1,34 +1,38 @@
+type typ_exp =
```

Comment:
I think we should adopt the convention that `Diff.x` represent the diff between two values of type `Types.x`. I'd rename this to `type_expr`.